### DISCUSSION TITLE
Dynamically render background color

### DISCUSSION THREAD

**Discussion by MeenaFoerderer** [neuefische/questions#365] [Web] [answered]
Hello,
I have an array of objects (180 "talks") and each object has a "session" property. There are 18 different sessions in total. I want that each session has a different background color.
I am now a bit stuck in the logic to assign different colors to different sessions. This is the code I have started so far. Everything works fine with the `sessionArray` and `colors` variables, but I don't know how to finish the function:

```
import randomColor from "randomcolor";

.....

 const sessionArray = Array.from(new Set(talks.map((talk) => talk.session)));
 const colors = randomColor({ count: sessionArray.length });

 let colorIndex = 0;

 function setSessionBgColors() {
    if (colorIndex > colors.length - 1) {
      colorIndex = 0;
    } else {
      colorIndex++;
    }
    sessionArray.forEach((session) => ????????????
    );
  }

  setSessionBgColors();
```
this is my actual PR, but the code above is currently not implemented: https://github.com/MeenaFoerderer/Capstone-project/pull/18

**Comment by actyralf** [comment, marked as answer]
You have to store the mapping between session name and bg-color somewhere. The most convenient way would be an object with the session name or session id as key, e.g.:
```
const sessionConfig={
   "session1": {
      bgColor: "#ff0000"
   },
   "session2": {
      bgColor: "#00ff00"
   }
}
```
If your sessions are fixed (i.e. you know them beforehand and they don't change) you could prepare this as static data, import it and just use it.
If sessions are dynamic you would have to create this object in state and update it when session data changes.
You could then use this mapping for rendering the bgColor of a specific talk, e.g.:

```
<div style={{backgroundColor: sessionConfig[talk.session].bgColor}}>
  // render talk data here
</div>
```
This would be the general concept, you would have to adapt this to your actual data structure...
